docs/refresh-token.md

```diff
@@ -0,0 +1,159 @@
+
+
+## Refreshing Access Token
+
+The following is an example of middleware that handles the token refresh process based on the sample provided by [@Good-man](https://github.com/Good-man) on this issue https://github.com/okta/okta-aspnet/issues/130.
+
+### Startup.cs
+
+```csharp
+        public void Configuration(IAppBuilder app)
+        {
+            app.SetDefaultSignInAsAuthenticationType(CookieAuthenticationDefaults.AuthenticationType);
+
+            app.UseCookieAuthentication(new CookieAuthenticationOptions());
+            
+            // Adding the Refresh token middleware
+            var oktaDomain = ConfigurationManager.AppSettings["okta:OktaDomain"];
+            var authServerId = ConfigurationManager.AppSettings["okta:AuthorizationServerId"];
+            var tokenEndpoint = $"{oktaDomain}/oauth2/{authServerId}/v1/token";
+
+            app.Use(typeof(RefreshTokenMiddleware), new RefreshTokenMiddlewareOptions
+            {
+                TokenEndpoint = tokenEndpoint,
+                ClientId = ConfigurationManager.AppSettings["okta:ClientId"],
+                ClientSecret = ConfigurationManager.AppSettings["okta:ClientSecret"],
+            });
+            
+            // Adding Okta Mvc
+            app.UseOktaMvc(new OktaMvcOptions()
+            {
+                OktaDomain = oktaDomain,
+                ClientId = ConfigurationManager.AppSettings["okta:ClientId"],
+                ClientSecret = ConfigurationManager.AppSettings["okta:ClientSecret"],
+                AuthorizationServerId = authServerId,
+                RedirectUri = ConfigurationManager.AppSettings["okta:RedirectUri"],
+                PostLogoutRedirectUri = ConfigurationManager.AppSettings["okta:PostLogoutRedirectUri"],
+                GetClaimsFromUserInfoEndpoint = true,
+
+                Scope = new List<string> { "openid", "profile", "email", "offline_access" },
+            });
+        }
+```
+
+
+### RefreshTokenMiddleware.cs
+
+```csharp
+    public class RefreshTokenMiddleware : OwinMiddleware
+    {
+        private readonly RefreshTokenMiddlewareOptions _options;
+
+        public RefreshTokenMiddleware(OwinMiddleware next, RefreshTokenMiddlewareOptions options) : base(next)
+        {
+            _options = options;
+        }
+
+        public override async Task Invoke(IOwinContext context)
+        {
+            if (AccessTokenExpired(context) && HasRefreshToken(context))
+            {
+                await RefreshTokens(context);
+            }
+
+            await Next.Invoke(context);
+        }
+
+        private bool HasRefreshToken(IOwinContext context)
+        {
+            if (context.Authentication.User.Identity is ClaimsIdentity id && id.IsAuthenticated)
+            {
+                return id.Claims.Any(c => c.Type == JwtClaimTypes.Name);
```

Comment:
Should we find a claim named "refresh_token" to determine if the user has a refresh token instead of `JwtClaimTypes.Name`?
Also, I suggest renaming `id` to `identity`.